src/engraving/dom/arpeggio.cpp

```diff
@@ -52,13 +54,140 @@ Arpeggio::Arpeggio(Chord* parent)
     m_userLen2 = 0.0;
     m_playArpeggio = true;
     m_stretch = 1.0;
+
+    parent->setSpanArpeggio(this);
+}
+
+Arpeggio::~Arpeggio()
+{
+    // Remove reference to this arpeggio in any chords it may have spanned
+    Chord* _chord = chord();
+    if (!_chord) {
+        return;
+    }
+    for (track_idx_t _track = track(); _track <= track() + m_span; _track++) {
+        EngravingItem* e = _chord->segment()->element(_track);
+        if (e && e->isChord() && toChord(e)->spanArpeggio() == this) {
+            toChord(e)->setSpanArpeggio(nullptr);
+        }
+    }
 }
 
 const TranslatableString& Arpeggio::arpeggioTypeName() const
 {
     return TConv::userName(m_arpeggioType);
 }
 
+void Arpeggio::findAndAttachToChords()
+{
+    Chord* _chord = chord();
+    track_idx_t strack = track();
+    track_idx_t etrack = track() + (m_span - 1);
+
+    for (track_idx_t track = strack; track <= etrack; track++) {
+        EngravingItem* e = _chord->segment()->element(track);
+        if (e && e->isChord()) {
+            toChord(e)->undoChangeSpanArpeggio(this);
+        }
+    }
+}
+
+void Arpeggio::detachFromChords(track_idx_t strack, track_idx_t etrack)
+{
+    Chord* _chord = chord();
+    if (!_chord) {
+        return;
+    }
+    for (track_idx_t track = strack; track <= etrack; track++) {
+        EngravingItem* e = _chord->segment()->element(track);
+        if (e && e->isChord() && toChord(e)->spanArpeggio() == this) {
+            toChord(e)->undoChangeSpanArpeggio(nullptr);
+        }
+    }
+}
+
+void Arpeggio::rebaseStartAnchor(int direction)
```

Comment:
It would be better to use enum instead of int (rebaseStartAnchor/rebaseEndAnchor)